ui/src/pages/Questions/Detail/components/Reactions/index.tsx

```diff
@@ -0,0 +1,146 @@
+import { FC, memo, useEffect, useState } from 'react';
+import { Button, OverlayTrigger, Tooltip } from 'react-bootstrap';
+import { useTranslation } from 'react-i18next';
+
+import classNames from 'classnames';
+
+import { Icon } from '@/components';
+import { queryReactions, updateReaction } from '@/services';
+import { tryNormalLogged } from '@/utils/guard';
+import { ReactionItem } from '@/common/interface';
+
+interface Props {
+  objectId: string;
+  showAddCommentBtn?: boolean;
+  handleClickComment: () => void;
+}
+
+const emojiMap = [
+  {
+    name: 'heart',
+    icon: 'heart-fill',
+    className: 'text-danger',
+  },
+  {
+    name: 'smile',
+    icon: 'emoji-laughing-fill',
+    className: 'text-warning',
+  },
+  {
+    name: 'frown',
+    icon: 'emoji-frown-fill',
+    className: 'text-warning',
+  },
+];
+
+const Index: FC<Props> = ({
+  objectId,
+  showAddCommentBtn,
+  handleClickComment,
+}) => {
+  const [reactions, setReactions] = useState<Record<string, ReactionItem>>();
+  const { t } = useTranslation('translation');
+
+  useEffect(() => {
+    queryReactions(objectId).then((res) => {
+      setReactions(res.reaction_summary);
+    });
+  }, []);
+
+  const handleSubmit = (params: { object_id: string; emoji: string }) => {
+    if (!tryNormalLogged(true)) {
+      return;
+    }
+    updateReaction({
+      ...params,
+      reaction:
+        reactions &&
+        reactions[params.emoji] &&
+        reactions[params.emoji].is_active
+          ? 'deactivate'
+          : 'activate',
+    }).then((res) => {
+      setReactions(res.reaction_summary);
```

Comment:
Add the following code to enable the tooltip to be closed after changing updateReaction
```
setReactIsActive(false);
```

Reply:
updated.